### PR TITLE
ppl: inline upstream patch due to 401 error

### LINKED ISSUE #57499: "brew install -s ppl" fails - 401 Unauthorized
# Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

brew reinstall -s ppl

## What happened (include command output)

<details>
  <summary>Command output</summary>
  <pre>
==> Downloading http://www.cs.unipr.it/git/gitweb.cgi?p=ppl/ppl.git;a=commitdiff_plain;h=c39f6a07b51f89e365b05ba4147aa2aa448febd7
==> Downloading from https://www.cs.unipr.it/git/gitweb.cgi?p=ppl/ppl.git;a=commitdiff_plain;h=c39f6a07b51f89e365b05ba4147aa2aa448febd7
#=#=-#  #                                                                     
curl: (22) The requested URL returned error: 401 Unauthorized
Error: Failed to download resource "ppl--patch"
Download failed: http://www.cs.unipr.it/git/gitweb.cgi?p=ppl/ppl.git;a=commitdiff_plain;h=c39f6a07b51f89e365b05ba4147aa2aa448febd7
  </pre>
</details>

## What you expected to happen

a clean installation

## Step-by-step reproduction instructions (by running `brew install` commands)

brew install -s ppl || brew reinstall -s ppl

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

~add homebrew mirror for source artifact~

use official url for source artifact

fixes #57499

### CODE REVIEW

**Comment by Bo98** [comment]
> add homebrew mirror for source artifact

Can we not use the official source?

https://www.bugseng.com/products/ppl/download/ftp/releases/1.2/ppl-1.2.tar.xz

**Comment by chenrui333** [comment]
yeah, definitely we should use the official one. good catch.

File: Formula/ppl.rb [hunk 1]
```diff
@@ -14,12 +16,12 @@ class Ppl < Formula
...
-  # Upstream commit, remove for next version
-  patch do
-    url "http://www.cs.unipr.it/git/gitweb.cgi?p=ppl/ppl.git;a=commitdiff_plain;h=c39f6a07b51f89e365b05ba4147aa2aa448febd7"
-    sha256 "6786f432784b74b81805b1d97e97cd1cc9f68653077681bb4f531466cbf8dc99"
-  end
+  # Fix build failure with clang 5+.
+  # https://www.cs.unipr.it/mantis/view.php?id=2128
+  # http://www.cs.unipr.it/git/gitweb.cgi?p=ppl/ppl.git;a=commit;h=c39f6a07b51f89e365b05ba4147aa2aa448febd7
+  # since 401 error on the `www.cs.unipr.it` links adopt the patch from macports
+  # patch reference, https://github.com/macports/macports-ports/commit/e5de9cc65a8e91fcbb9a3d90911569169f0ccf88
```
Comment (SMillerDev) [inline Formula/ppl.rb:23]: Why not just use that as a patch then?

Comment (jonchang) [reply Formula/ppl.rb:23]: We would have to patch the patch as our paths vary from MacPorts's 😅

Comment (chenrui333) [reply Formula/ppl.rb:23]: yeah, macports has its own installation, so I added the patch separately in here.

**Review by SMillerDev — DISMISSED** [review]

**Review by jonchang — APPROVED** [review]

**Pull request closed without merging**